src/functions.js

```diff
@@ -0,0 +1,114 @@
+// select variables
+const todoListElement = document.getElementById('todo');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+const todoInput = document.querySelector('input');
+let EditTodoId = -1;
+// first render
+
+// save Todo function to the local storage of a user machine
+/* eslint-disable consistent-return */
+export function saveTodo() {
+  const todoValue = todoInput.value;
+  const isEmpty = todoValue === '';
+  // check duplicate input
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+  // check if userenters an  empty input and clicks
+  /* eslint-disable no-else-return */
+  /* eslint-disable brace-style */
```

Comment:
- [ ] It is not acceptable to disable any linters rule because you are just giving room for some bugs that may arise in the future or now due to syntax errors that you may not be aware of due to the complexity of the project. We should focus on correcting errors marked by linters instead of disabling them so please do not disable any rule like here.